app/models/post.rb

```diff
@@ -0,0 +1,12 @@
+class Post < ApplicationRecord
+  belongs_to :users
+  has_many :comments
+  has_many :likes
+  def update_post_count
+    author.increment!(:posts_counter)
```

Comment:
- [ ] Using `author.increment!(:posts_counter)` would throw a runtime error hence your app will crash due to using the `author` field as it doesn't appear anywhere on your model or migration, I would suggest you to refactor line 2 as follows: `belongs_to :author, class_name: 'User'` in order to have the line 6 working, otherwise you will need to use a different approach
- [ ] [Optional] You can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to database, then make the `update_post_counter` method private